hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -354,8 +363,39 @@ private void onStateInitialized(
             @NonNull final SwirldDualState dualState,
             @NonNull final InitTrigger trigger,
             @Nullable final SoftwareVersion previousVersion) {
+        // Initialize the configuration from disk. We must do this BEFORE we run migration, because the various
+        // migration methods may depend on configuration to do their work. For example, the token service migration code
+        // needs to know the token treasury account, which has an account ID specified in config. The initial config
+        // file in state, created by the file service migration, will match what we have here, so we don't have to worry
+        // about re-loading config after migration.
+        logger.info("Initializing configuration with trigger {}", trigger);
+        switch (trigger) {
+            case EVENT_STREAM_RECOVERY -> configProvider =
```

Comment:
Why is this case different from `RESTART, RECONNECT`?

Reply:
Good question. The code initializing this config provider was previously instantiated with `new ConfigProviderImpl(false)`, so maybe that's already sufficient and the extra config constructor isn't needed. I'll ask around.